grammar.js

```diff
@@ -53,7 +53,7 @@ module.exports = grammar({
                 ),
             ),
 
-        int: $ => /-?([0-9][0-9_]*|0x[0-9A-Fa-f][0-9A-Fa-f_]*)/,
+        int: $ => seq(optional('$'), /-?([0-9][0-9_]*|0x[0-9A-Fa-f][0-9A-Fa-f_]*)/),
```

Comment:
I'm assuming the `$` sign is another way to write hex literals? In that case I think it should be part of the regex as an alternative to the `0x` prefix so that for example `$0xff` and `$-10` aren't considered valid.
```suggestion
        int: $ => /-?([0-9][0-9_]*|(0x|\$)[0-9A-Fa-f][0-9A-Fa-f_]*)/,
```

Reply:
Makes sense!